### PR TITLE
Implement bounding boxes for center crop

### PR DESCRIPTION
**Pull request by AmedeoBiolatti**
# What does this PR do?

Implement # (https://github.com/keras-team/keras-cv/issues/731)

centercrop

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

@LukeWood and @tanzhenyu

### CODE REVIEW

File: keras_cv/layers/preprocessing/center_crop.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+from tensorflow import keras
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class CenterCrop(BaseImageAugmentationLayer):
+    """A preprocessing layer which crops images.
+
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing/center_crop.py:27]: This layers crops the central portion of the images to a target size. If an image is smaller than the target size, it will be resized and cropped so as to return the largest possible window in the image that matches the target aspect ratio.

File: keras_cv/layers/preprocessing/center_crop.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            image=None,
+            label=None,
+            bounding_boxes=None,
+            keypoints=None,
+            segmentation_mask=None,
+    ):
+        image_shape = tf.shape(image)
+        return image_shape[-3], image_shape[-2]
+
+    def _transform_bounding_boxes(self, bounding_boxes, transformation):
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing/center_crop.py:79]: do you need to filter boxes that has no overlap with the center crop region?

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/layers/preprocessing/center_crop.py (hunk 2 above)
Comment (AmedeoBiolatti) [reply keras_cv/layers/preprocessing/center_crop.py:81]: This is handled by `bounding_box.clip_to_image` inside `augment_bounding_boxes,` as seen in the unit test it seems to work

File: keras_cv/layers/preprocessing/center_crop.py (hunk 1 above)
Comment (bhack) [reply keras_cv/layers/preprocessing/center_crop.py:29]: >  it will be resized..

Do we need to introduce some safety warning log about the ratio? As It could happen to have extreme values processing bulk datasets...

Comment (bhack) [reply keras_cv/layers/preprocessing/center_crop.py:29]: ...Or at least a explicit flag to enable these cases..

File: keras_cv/layers/preprocessing/center_crop.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/center_crop.py:83]: Yep!  clip_to_image() does in fact filter out of bounds bounding boxes!

**Comment by LukeWood** [comment]
make sure to add entries to:

- serialization_test
- with_labels_test

File: keras_cv/layers/preprocessing/center_crop.py (hunk 1 above)
Comment (AmedeoBiolatti) [reply keras_cv/layers/preprocessing/center_crop.py:29]: Is it possible to have more details? Should be raise a warning when the aspect ratio is too much different or when the size is too different?

File: keras_cv/layers/preprocessing/center_crop.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/layers/preprocessing/center_crop.py:83]: you might also need to add a call to `bounding_box.filter_sentinels()` to go from dense -1 padded tensor -> ragged

**Comment by LukeWood** [comment]
Here's a thought: CenterCrop is not really ideal for bounding boxes because you can literally crop entire labels out.
Perhaps we should intentionally not support it, and just raise an informative error message telling users to use Resizing() instead?


@tanzhenyu thoughts?

**Comment by divyashreepathihalli** [comment]
> 

@LukeWood - I think that would be the case for random crop  and random resize crop as well. I think there is value is supporting augment bounding boxes for this with a warning. WDYT?

**Comment by LukeWood** [comment]
> > 
> 
> @LukeWood - I think that would be the case for random crop and random resize crop as well. I think there is value is supporting augment bounding boxes for this with a warning. WDYT?

@AmedeoBiolatti lets add a warning for center crop with bounding boxes indicating that this may be causing many boxes to be dropped.

**Comment by AmedeoBiolatti** [comment]
> @AmedeoBiolatti lets add a warning for center crop with bounding boxes indicating that this may be causing many boxes to be dropped.

@LukeWood  Done! Now I will try to resolve the conflicts

**Comment by AmedeoBiolatti** [comment]
@LukeWood the conflicts are solved, let me know if anything else is needed

**Review by LukeWood — CHANGES_REQUESTED** [review]
Lets remove the box portion of this PR and just merge CenterCrop as a KerasCV layer for now.

**Comment by divyashreepathihalli** [comment]
closing because this is stale.
